homeassistant/components/ezviz/switch.py

```diff
@@ -24,61 +132,75 @@ async def async_setup_entry(
         DATA_COORDINATOR
     ]
 
-    supported_switches = {switches.value for switches in DeviceSwitchType}
-
     async_add_entities(
         [
-            EzvizSwitch(coordinator, camera, switch)
+            EzvizSwitch(coordinator, camera, switch_number)
             for camera in coordinator.data
-            for switch in coordinator.data[camera].get("switches")
-            if switch in supported_switches
+            for switch_number in coordinator.data[camera]["switches"]
+            if switch_number in SWITCH_TYPES
+            if SWITCH_TYPES[switch_number].supported_ext
+            in coordinator.data[camera]["supportExt"]
+            or SWITCH_TYPES[switch_number].supported_ext is None
         ]
     )
 
 
 class EzvizSwitch(EzvizEntity, SwitchEntity):
     """Representation of a EZVIZ sensor."""
 
-    _attr_device_class = SwitchDeviceClass.SWITCH
+    _attr_has_entity_name = True
 
     def __init__(
-        self, coordinator: EzvizDataUpdateCoordinator, serial: str, switch: str
+        self, coordinator: EzvizDataUpdateCoordinator, serial: str, switch_number: int
     ) -> None:
         """Initialize the switch."""
         super().__init__(coordinator, serial)
-        self._name = switch
-        self._attr_name = f"{self._camera_name} {DeviceSwitchType(switch).name.title()}"
+        self._switch_number = switch_number
         self._attr_unique_id = (
-            f"{serial}_{self._camera_name}.{DeviceSwitchType(switch).name}"
+            f"{serial}_{self._camera_name}.{DeviceSwitchType(switch_number).name}"
```

Comment:
Really possible to just switch from a switch name to a number?

Reply:
The data type was wrong initially and the name of the variable.

Data structure is genuinely int, DeviceSwitchType is a enum class with number to name mappings.

Actual data from one of my cameras:
{1: True, 2: False, 3: True, 7: False, 8: True, 10: True, 15: False, 21: False, 22: True, 25: False, 29: False, 31: False, 32: False, 200: False, 202: False, 300: False, 301: False, 302: False, 303: False, 304: False, 305: False, 604: False, 650: False, 651: False, 700: False, 701: False, 702: False}